### PR TITLE
MF-387: Patient chart summary tile states - error state

### PR DESCRIPTION
**Pull request by denniskigen**
[Ticket](https://issues.openmrs.org/browse/MF-387)

error-state

This PR adds an `ErrorState` component that can be used in the patient chart widgets to communicate to the user the root cause when data cannot be surfaced for some reason. 

I think the spirit behind Ciaran's design was to make the information displayed as simple as possible for the user to consume whilst still being helpful. A few concerns surface however, the information captured here is too generic, for example. He proposes having a few preset illustrations + custom error messages for the most common error codes to begin with. I appreciate any thoughts on this.

Here's the actual error message behind the failed immunizations request:
immunizations-error

### CODE REVIEW

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks again @denniskigen ! :beers:

File: src/ui-components/error-state/error-state.component.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const { t } = useTranslation();
+
+  return (
+    <Tile light>
+      <h1 className={styles.heading}>{props.headerTitle}</h1>
+      <ErrorIllustration />
+      <p className={styles.content}>
+        <Trans i18nKey="errorStateText">Sorry, there was an error</Trans>
+      </p>
+      <p className={styles.errMessage}>Database does not exist for this tile</p>
```
Comment (brandones) [inline src/ui-components/error-state/error-state.component.tsx]: This message should be a prop, right?

**Review by brandones — APPROVED** [review]
Looks good. I left a comment [on Zeplin](https://app.zeplin.io/project/5f7223cfda10f94d67cec6d0/screen/5faaa506b332b96ffebb0272) asking Ciaran what's the plan with the error illustration, which presumably should not always be "404."

File: src/ui-components/error-state/error-state.component.tsx (hunk 1 above)
Comment (denniskigen) [reply src/ui-components/error-state/error-state.component.tsx]: Indeed it should. I left it as so pending clarification about exactly where this message should come from and exactly what it should look like. See PR description above for a bit on Ciaran's thoughts.

**Review by donaldkibet — APPROVED** [review]
🚀

**Comment by gracepotma** [comment]
@brandones I just followed up with Ciaran; he hadn't seen the Zeplin comment, but he'll look into it shortly. In the meantime we'll need to come up with some user-facing content for those error messages. I can have a think on this

**Comment by denniskigen** [comment]
Screenshot 2020-12-08 at 12 51 04

@FlorianRappl I have updated the tile to match the latest designs. Presently, it's only configured to work for the immunizations widget. I could incorporate the error handling logic for the other widgets here or in a future PR. Presently, I'm leaning towards merging this as is and working on the other widgets (at least the vitals and biometrics ones) later on.

File: src/ui-components/error-state/error-state.component.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  return (
+    <Tile light>
+      <h1 className={styles.heading}>{headerTitle}</h1>
+      <p className={styles.errorMessage}>
+        Error {`${error.response.status}: `}
+        {error.response.statusText}
+      </p>
+      <p className={styles.errorCopy}>
+        Sorry, there was a problem displaying this information. You can try to
```
Comment (FlorianRappl) [inline src/ui-components/error-state/error-state.component.tsx]: Can we make this translatable?

**Review by FlorianRappl — COMMENTED** [review]
Looks good - but at one translation for the text please.

File: src/ui-components/error-state/error-state.component.tsx (hunk 2 above)
Comment (denniskigen) [reply src/ui-components/error-state/error-state.component.tsx]: Yes!

Linked PR: https://github.com/openmrs/openmrs-esm-patient-chart/pull/196.

Screenshot 2020-12-08 at 13 16 11

**Review by FlorianRappl — APPROVED** [review]
Great, thanks :beers:!

**Comment by gracepotma** [comment]
Looks much better @denniskigen, very clear and easier to implement. 🚀 👍

**Pull request merged**